internal/services/search/search_service_resource.go

```diff
@@ -161,46 +216,219 @@ func resourceSearchServiceCreateUpdate(d *pluginsdk.ResourceData, meta interface
+			expandedIdentity, err := identity.ExpandSystemAssigned(d.Get("identity").([]interface{}))
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+
+			// NOTE: Passing type 'None' on update will remove all identities from the service.
+			model.Identity = expandedIdentity
+		}
+
+		if d.HasChange("hosting_mode") {
+			hostingMode := services.HostingMode(d.Get("hosting_mode").(string))
+			if pointer.From(model.Sku.Name) != services.SkuNameStandardThree && hostingMode == services.HostingModeHighDensity {
+				return fmt.Errorf("'hosting_mode' can only be set to 'highDensity' if the 'sku' is 'standard3', got %q", pointer.From(model.Sku.Name))
+			}
+
+			model.Properties.HostingMode = pointer.To(hostingMode)
+		}
+
+		if d.HasChange("cmk_enforcement_enabled") {
+			cmkEnforcement := services.SearchEncryptionWithCmkDisabled
+			if enabled := d.Get("cmk_enforcement_enabled").(bool); enabled {
+				cmkEnforcement = services.SearchEncryptionWithCmkEnabled
+			}
+
+			model.Properties.EncryptionWithCmk.Enforcement = pointer.To(cmkEnforcement)
+		}
+
+		apiAccessControl := d.Get("api_access_control").(*pluginsdk.Set).List()
+
+		if d.HasChange("api_access_control") {
+			authenticationOptions, err := expandSearchServiceAuthOptions(apiAccessControl)
+			if err != nil {
+				return fmt.Errorf("expanding 'api_access_control': %+v", err)
+			}
+
+			// If 'authenticationOptions' are nil it is RBAC only mode...
+			disableLocalAuth := false
+			if authenticationOptions == nil {
+				disableLocalAuth = true
+			}
+
+			model.Properties.DisableLocalAuth = pointer.To(disableLocalAuth)
+			model.Properties.AuthOptions = authenticationOptions
+		}
+
+		if d.HasChange("replica_count") {
+			replicaCount, err := validateSearchServiceReplicaCount(int64(d.Get("replica_count").(int)), pointer.From(model.Sku.Name))
+			if err != nil {
+				return err
+			}
+
+			model.Properties.ReplicaCount = pointer.To(replicaCount)
+		}
+
+		if d.HasChange("partition_count") {
+			partitionCount := int64(d.Get("partition_count").(int))
+			// NOTE: 'partition_count' values greater than 1 are not valid for 'free' or 'basic' SKUs...
+			if (pointer.From(model.Sku.Name) == services.SkuNameFree || pointer.From(model.Sku.Name) == services.SkuNameBasic) && partitionCount > 1 {
+				return fmt.Errorf("'partition_count' values greater than 1 cannot be set for the %q SKU, got %d)", pointer.From(model.Sku.Name), partitionCount)
+			}
+
+			// NOTE: If SKU is 'standard3' and the 'hosting_mode' is set to 'highDensity' the maximum number of partitions allowed is 3
+			// where if 'hosting_mode' is set to 'default' the maximum number of partitions is 12...
+			if pointer.From(model.Sku.Name) == services.SkuNameStandardThree && partitionCount > 3 && pointer.From(model.Properties.HostingMode) == services.HostingModeHighDensity {
+				return fmt.Errorf("'standard3' SKUs in 'highDensity' mode can have a maximum of 3 partitions, got %d", partitionCount)
```

Comment:
I was told by another reviewer, in a different PR, that they wanted me to use the hard coded strings instead of the constants since it made reading the error string harder to ascertain during the review. Is there a HashiCorp coding style standard for this? This back and forth depending on the reviewer is a bit confusing as a contributor.

Reply:
Fixed.